domain/application/state/application.go

```diff
@@ -876,3 +926,55 @@ AND a.name = $applicationName.name
 	}
 	return result, nil
 }
+
+// RemoveUnitMaybeApplication removes the unit from state, and may remove
+// its application as well, if the application is Dying and no other references
+// to it exist. It will fail if the unit is not Dead.
+// An error satisfying [applicationerrors.UnitNotFound] is returned
+// if the unit is not found.
+func (st *ApplicationState) RemoveUnitMaybeApplication(ctx domain.AtomicContext, unitName string) error {
+	unit := unitID{Name: unitName}
+	peerCountQuery := `
+SELECT a.life_id as &unitCount.app_life_id, u.life_id AS &unitCount.unit_life_id, u.uuid, count(peer.uuid) AS &unitCount.count
```

Comment:
Case `as`. Does the u.uuid automatically assigned to unitCount.uuid? Or do we actually need to select the u.uuid?

Reply:
as, this was not meant to be there